bitcoin/src/blockdata/constants.rs

```diff
@@ -47,7 +47,7 @@ pub const MAX_STACK_ELEMENT_SIZE: usize = 520;
 /// How may blocks between halvings.
 pub const SUBSIDY_HALVING_INTERVAL: u32 = 210_000;
 /// Maximum allowed value for an integer in Script.
-pub const MAX_SCRIPTNUM_VALUE: u32 = 0x80000000; // 2^31
+pub const MAX_SCRIPTNUM_VALUE: u64 = 0x7FFFFFFFFF; // 2^39 - 1
```

Comment:
I think the history is from when rust-miniscript implemented locktime.
https://github.com/rust-bitcoin/rust-miniscript/pull/408#discussion_r917464026

Reply:
I think we should deprecate this constant (and not change it to reduce breakage). It's such a weird thing. Elsewhere I have advocated having `push_locktime` method(s) to push locktimes, which as you note, have a different maximum value than every other scriptnum.

Another possible, though unlikely, interpretation of this constant is that it could be the maximum *output* of an arithmetic opcode. This, I think, is 2^31 - 1.

Also it's weird that we have a maximum value represented as a u32, and no minimum value.

My feeling is that if you are doing something that cares about this constant, you should define your own constants that clearly represent whatever it is you mean. Good find @ChrisCho-H that this constant originates in Miniscript doing something fairly specific, propagating poorly-documented constants into this library, and apparently getting it wrong.

Reply:
I agree with @apoelstra. It'd be better not to expose this weird constant. Overflow can be checked internally.

Reply:
I've just force pushed 9620cdf, removing this constant rather than changing its value.